src/api/app/models/staging/staged_requests.rb

```diff
@@ -84,6 +85,44 @@ def requests
     staging_workflow.unassigned_requests.where(number: request_numbers)
   end
 
+  def link_package(bs_request_action)
+    request = bs_request_action.bs_request
+
+    source_package = Package.get_by_project_and_name!(bs_request_action.source_project,
+                                                      bs_request_action.source_package)
+
+    # we need the revision and md5 and make it sticky in the _link
+    # packages in staging a locked in a specific version
+    backend_package_information = Directory.hashed(project: source_package.project.name,
+                                                   package: source_package.name, expand: 1)
```

Comment:
we're getting closer. But let me repeat what I wrote in chat: you forgot the (optional) revision

Reply:
And I'd suggest you just use `source_package.dir_hash`